src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -127,15 +127,22 @@ bool Series::saveToFolder(const std::string& folder) const
 
     bool ret = true;
 
-    auto checkWrite = [&ret, &folder](const std::string& name, const std::vector<double>& content) {
-        ret = writeVectorToFile(folder + SEP + name, content) && ret;
+    auto checkWrite = [&ret, &folder]
+        (const std::string& filename, const std::vector<double>& timeSeries, double defValue)
+    {
+        // if vector is only default values we don't save it
+        if (all_of(timeSeries.begin(), timeSeries.end(), [&defValue] (const double& i)
+                    { return i == defValue; }))
+            return;
+
+        ret = writeVectorToFile(folder + SEP + filename, timeSeries) && ret;
```

Comment:
Please use std::find to slightly simplify this. Also, you don't need `return`, just add `!` in the `if` block